File: python/e786.py
# -*- coding: utf-8 -*-
"""
Created on Tue Feb 18 00:45:03 2020

@author: Username
"""

import sys

for s in [input()]:
    if len(s)%2 != 0: 
        print('NO')
        continue
    num=len(s)//2
    tes=0
    #print(s[:num])
    #print(s[:num-1:-1])
    if s[:num] == s[:num-1:-1]: tes=1
    if tes==1:
        print('YES')
        print(s[:num])
    else: print('NO')

File: c++/e786.cpp
#include <iostream>
#include <cstring>
using namespace std;

int main(){
    string s;
    while (cin >> s) {
        int num{s.size()};
        if (num%2) cout << "NO\n";
        else{
            bool ok{true};
            string str="";
            for (int i = 0; i < num/2 && ok;i++){
                if (s[i] != s[num-i-1]) ok = false;
                str+=s[i];
            }
            if (ok) cout << "YES\n" << str << "\n";
            else cout << "NO\n";
        }
    }
}